sysl2/sysl/swagger/types.go

```diff
@@ -12,14 +12,51 @@ import (
 	"github.com/go-openapi/spec"
 )
 
-type Type struct {
-	Name       string
+type Type interface {
+	Name() string
+}
+
+type StandardType struct {
+	name       string
 	Properties []Field
-	isAlias    bool
-	isArray    bool
-	isEnum     bool
 }
 
+func (s *StandardType) Name() string { return s.name }
+
+type SyslBuiltIn struct {
+	name string
+}
+
+func (s *SyslBuiltIn) Name() string { return s.name }
+
+type Alias struct {
+	name  string
+	Alias Type
+}
+
+func (s *Alias) Name() string { return s.name }
+func NewStringAlias(name string) Type {
+	return &Alias{
+		name:  name,
+		Alias: &SyslBuiltIn{name: "string"},
+	}
+}
+
+type Array struct {
+	name  string
+	Items Type
+}
+
+func (s *Array) Name() string { return s.name }
+
+// sysl doesnt really support enums yet but leave for future use
+type Enum struct {
```

Comment:
If not currently in use, should omit.

Reply:
fair enough, Removed the Item slice, kept the struct. It would be good to at least add the possible values as documentattion to the generated sysl in the future